### ISSUE TITLE
preg_match_all for iframe is not reliable

### ISSUE THREAD

**Issue by ajoah** [Really-Simple-Plugins/complianz-gdpr#320]
Hi,

I found a strange issue on iframe parsing (for replace iframe src for a youtube video for example) : https://github.com/Really-Simple-Plugins/complianz-gdpr/blob/f5abe46ef920ed4f4792c031703db80f890c59a1/class-cookie-blocker.php#L231-L232

On one of my website, the youtube placeholder didn't work. Indeed, the `preg_match_all` return false.

After investigation, it seems that `preg_match_all` is not reliable on long strings : https://stackoverflow.com/a/27868983/1659617

Maybe you would have to optimize the regex or get just the iframes in first time then in the loop parse the content of the iframe

Code to reproduce : 

```
<?php
$iframe_pattern = '/<(iframe)[^>].*?src=[\'"](http:\/\/|https:\/\/|\/\/)([\w.,;@?^=%&:()\/~+#!\- *]*?)[\'"].*?>.*?<\/iframe>/is';
$html = 'html content'
if ( preg_match_all( $iframe_pattern, $html, $matches, PREG_PATTERN_ORDER ) ) {
  echo 'ok';
}else{
  echo 'ko';
}
```
If you have a private way, i can send you my html code to reproduce the issue in PHP 7.4.
I am on Wordpress slack, username => ajoah

**Comment by ajoah** [comment]
Hi @rlankhorst,

Yes it works with this regex 👍

**Comment by rlankhorst** [comment]
@ajoah great, thanks for checking. This won't make the upcoming release, but will be included in the 6.0 update, coming soon.
